interfaces/apparmor/spec.go

```diff
@@ -567,6 +567,64 @@ func snippetFromLayout(layout *snap.Layout) string {
 	return fmt.Sprintf("# Layout path: %s\n# (no extra permissions required for symlink)", mountPoint)
 }
 
+// emitEnsureDir creates an apparmor snippet that permits snap-update-ns to create
+// missing directories for the calling user according to the provided ensure directory spec.
+// This function is currently used as counterpart for AddUserEnsureDirs, but can also be used
+// for permitting non-user ensure directory specs.
+func emitEnsureDir(spec *Specification, ifaceName string, ensureDirSpec *interfaces.EnsureDirSpec) {
+	ensureDir := ensureDirSpec.EnsureDir
+	mustExistDir := ensureDirSpec.MustExistDir
```

Comment:
should we check that mustExistDir is a prefix of ensureDir ?

Reply:
As discussed, we do error checking on mount side. This algo already [iterates to after `mustExistDir`](https://github.com/snapcore/snapd/pull/13260/files#diff-78692e23a83df81d72626c6a2a4cb7d54281f8fbaa33073bf019555fe760500aR602), so will just not add anything in this case.